Web/Scotty/Internal/Types.hs

```diff
@@ -74,9 +74,10 @@ addRoute r s@(ScottyState {routes = rs}) = s { routes = r:rs }
 addHandler :: ErrorHandler e m -> ScottyState e m -> ScottyState e m
 addHandler h s = s { handler = h }
 
-newtype ScottyT e m a = ScottyT { runS :: State (ScottyState e m) a }
-    deriving ( Functor, Applicative, Monad )
+newtype ScottyT' e m n a = ScottyT { runS :: StateT (ScottyState e m) n a }
+    deriving ( Functor, Applicative, Monad, MonadTrans )
 
+type ScottyT e m = ScottyT' e m m
```

Comment:
This is technically a breaking change, as it's possible that current `scotty` code will fail to compile after this change due to `ScottyT` becoming a type synonym. For instance, you would no longer be able to define instances for `ScottyT` without enabling the use of `TypeSynonymInstances`, or another language extension which implies it. This is fine, but I would argue that this should come with a major version bump, and definitely a mention in the changelog.